pkg/rtc/mos.go

```diff
@@ -0,0 +1,60 @@
+package rtc
+
+import (
+	"github.com/livekit/protocol/livekit"
+)
+
+// MOS score calculation is based on webrtc-stats
+// available @ https://github.com/oanguenot/webrtc-stats
+
+const (
+	rtt = 10
+)
+
+func score2Rating(score float64) livekit.ConnectionQuality {
+	if score > 4.00 {
+		return livekit.ConnectionQuality_EXCELLENT
+	}
+
+	if score > 3.5 {
+		return livekit.ConnectionQuality_GOOD
+	}
+	return livekit.ConnectionQuality_POOR
+}
+
+func mosAudioEmodel(cur *ConnectionStat) float64 {
+
+	if cur == nil {
+		return 0.0
+	}
+
+	percentageLost := FixedPointToPercent(cur.FractionLost)
```

Comment:
Rather than taking max fraction lost, recommend calculating number of packets and number of packets lost and doing a percentage. Looks like the score is calculated once every 5 seconds. Video could send a lot of RTCP receiver reports in that window. Each of those reports could have number of packets like 10 - 15 and a loss of a couple of packets in one of those reports will have a fractional loss of over 10%. If every other report in the calculation window had no loss, the actual loss could be well under 1%, but that one sample could skew the reading.

Reply:
reading from buffer.stats and find delta percentage lost in the window